app/dao/mongo.go

```diff
@@ -0,0 +1,84 @@
+package dao
+
+import (
+	"fmt"
+
+	"github.com/companieshouse/payment-reconciler/app/config"
+	"github.com/companieshouse/payment-reconciler/app/models"
+	"github.com/globalsign/mgo"
+	"gopkg.in/mgo.v2/bson"
+)
+
+// Mongo provides a MongoDB implementation of the DAO
+type Mongo struct {
+	Config *config.Config
+}
+
+// New returns a new Mongo struct using the provided config
+func New(cfg *config.Config) *Mongo {
+
+	return &Mongo{
+		Config: cfg,
+	}
+}
+
+// getMongoSession retrieves a fresh MongoDB session
+func getMongoSession(cfg *config.Config) (*mgo.Session, error) {
```

Comment:
Could we follow a similar structure please ?
https://github.com/companieshouse/payments.api.ch.gov.uk/blob/develop/dao/mongo.go#L20

Reply:
The function should be private; it's used to return a new Mongo session and only `mongo.go` should have knowledge of that.

Reply:
https://github.com/companieshouse/payments.api.ch.gov.uk/blob/develop/dao/mongo.go#L20 seems to be private and returns a mongo session

Reply:
Followed logic from payments.api